src/web/common.rkt

```diff
@@ -122,18 +168,30 @@
   (define versions
     (reap [sow]
       (for ([(lang record) (in-dict languages)])
-        (match-define (list ext converter) record)
+        (match-define (list ext converter preprocess) record)
         (when (and (fpcore? in-prog*)
                    (or (not out-prog*) (fpcore? out-prog*))
                    (or (equal? ext "fpcore")                           
                        (and (supported-by-lang? in-prog* ext) ; must be valid in a given language  
                             (or (not out-prog*)
                                 (supported-by-lang? out-prog* ext)))))
           (define name (if identifier (symbol->string identifier) "code"))
-          (sow
-            (cons lang
-              (cons (converter in-prog* name)
-                    (and out-prog* (converter out-prog* name)))))))))
+          (define in (converter in-prog* name))
+          (define out (and out-prog* (converter out-prog* name)))
+          (define preprocess-lines
+            (string-join (map preprocess preprocesses) "\n" #:after-last "\n"))
+          (define out-with-sort-preprocessing
+            (if (equal? lang "TeX")
```

Comment:
Let's move this into a special TeX helper function with the same signature as `string-append`. That way, you can do:

```
(define combine-lines (if (equal? lang "TeX") combine-tex-preprocess string-append))
(combine-lines preprocess-lines output)
```